src/App.tsx

```diff
@@ -1,203 +1,280 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
 import { USER_ID } from './api/todos';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+import * as postService from './api/todos';
 
 export const App: React.FC = () => {
+  const [data, setData] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [filter, setFilter] = useState<'All' | 'Active' | 'Completed'>('All');
+  const [newTodoTitle, setNewTodoTitle] = useState<string>('');
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+
+  useEffect(() => {
+    setErrorMessage(null);
+    getTodos()
+      .then(setData)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    if (errorMessage) {
+      const timer = setTimeout(() => {
+        setErrorMessage(null);
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [errorMessage]);
+
+  const filteredTodos = data.filter(todo => {
+    if (filter === 'Active') {
+      return !todo.completed;
+    } else if (filter === 'Completed') {
+      return todo.completed;
+    }
+
+    return true;
+  });
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  function createTodo() {
+    if (newTodoTitle.trim() === '') {
+      setErrorMessage('Title should not be empty');
+
+      return;
+    }
+
+    const newTodoData = {
+      userId: USER_ID,
+      title: newTodoTitle.trim(),
+      completed: false,
+    };
+
+    setIsSubmitting(true);
+    postService.createTodo(newTodoData)
+      .then(newTodo => {
+        setData(currentTodos => [...currentTodos, newTodo]);
+        setNewTodoTitle('');
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+      })
+      .finally(() => {
+        setIsSubmitting(false);
+      });
+  }
+
+  function deleteTodo(id: number) {
+    setLoading(true);
+
+    postService.deleteTodo(id)
```

Comment:
why postService , we haven't posts in this task
clientService or apiService ot todosService suits better